### PR TITLE
Extend source with new models

### PR DESCRIPTION
**Pull request by zsvizi**
- refactored the files

### CODE REVIEW

**Comment by zsvizi** [comment]
Refactor the code to avoid code copy.

File: src/plotter.py [hunk 1]
```diff
@@ -11,12 +11,29 @@
...
 
 class Plotter:
-    def __init__(self, data, n_age: int) -> None:
+    def __init__(self, data, n_age: int, model: str) -> None:
         self.data = data
         self.n_age = n_age
-        self.labels = ["0-4", "5-9", "10-14", "15-19", "20-24", "25-29",
-                       "30-34", "35-39", "40-44", "45-49", "50-54", "55-59",
-                       "60-64", "65-69", "70-74", "75+"]
+        if model in ["rost", "kenya"]:
```
Comment (zsvizi) [inline src/plotter.py:17]: Create a JSON file where you put these lists, and load that JSON, and based on model name you can simply fill `self.labels`.

File: src/static/dataloader.py [hunk 2]
```diff
@@ -1,56 +1,112 @@
...
 PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
 
 
 class DataLoader:
-    def __init__(self):
+    def __init__(self, model: str):
+        self.model = model.lower()  # Standardize country string
+
+        # Dictionary mapping for the models and corresponding data files
+        model_data_files = {
```
Comment (zsvizi) [inline src/static/dataloader.py:14]: Create a JSON file and put this dictionary there. Load the JSON file and this structure will be available easily.

**Comment by zsvizi** [comment]
Now we reached the point, that it is a joke to store that many data files on Github. 
Follow these steps:
1. You have to remove the data files and put them into a GDrive folder. This would be good: https://drive.google.com/drive/folders/1kZrOSpC0W7lpfh22dR7I1EAz_hgcUaT7?usp=drive_link
2. Use package `gdown` to download the files. The download has to be called in the constructor of the class `DataLoader`. 
3. Don't forget to write a simply `if` to check, whether the files are already downloaded. E.g. use this code: `if not os.path.isfile(file_path): ...`.
4. Put the newly created JSON files also to the GDrive.

File: src/static/cm_leaf_preparator.py [hunk 3]
```diff
@@ -28,12 +29,16 @@ def _calculate_full_orig_cm(self) -> torch.Tensor:
         Returns: torch.Tensor: The full contact matrix.
         """
         contact_data = self.data.contact_data
```
Comment (zsvizi) [inline src/static/cm_leaf_preparator.py:31]: Create a `CMData` class, which loads the current model's contact matrix and calculates the full version. Then this logic will be there, not here.

File: src/static/dataloader.py [hunk 4]
```diff
@@ -1,56 +1,183 @@
 import json
 import os
-
 import torch
 import xlrd
+from gdown import download
+
+# Google Drive file links for various models
+model_ids = {
```
Comment (zsvizi) [inline src/static/dataloader.py:8]: This dict has to go to a JSON file, which is downloaded in the beginning (the link of that file can be hardcoded, since no other will be later).

**Comment by zsvizi** [comment]
Create classes `VMatrixCalculatorBase` and an `NGMCalculatorBase`, where you put the code included in the model-specific classes. Then inherit the model-specific classes from this base class. This will avoid real copy+pastes in the code.

File: src/static/dataloader.py [hunk 5]
```diff
@@ -1,56 +1,183 @@
...
+    def _set_file_paths(self):
+        """Set file paths for the selected model."""
+        data_files = {
+            "model_parameters": f"{self.model}_model_parameters.json",
+            "contact_matrices": f"{self.model}_contact_matrices.xls",
+            "age_distribution": f"{self.model}_age_distribution.xls",
+            "labels": "labels.json"
+        }
+
         self._model_parameters_data_file = os.path.join(PROJECT_PATH, "../data",
```
Comment (zsvizi) [inline src/static/dataloader.py:83]: Break this and the following 3 definitions after `os.path.join(`. In this case, the arguments will be aligned to each other.

**Comment by zsvizi** [comment]
One of the comments has to be applied for every model.
